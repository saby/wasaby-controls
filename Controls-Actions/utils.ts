import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
