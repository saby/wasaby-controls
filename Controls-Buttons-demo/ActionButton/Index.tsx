import ButtonType from 'Controls-Buttons-meta/buttonConnectedButtonType';
import { Button } from 'Controls-Buttons/buttonConnected';
import { PropsDemoEditor } from 'Controls-demo/ObjectTypeEditor/PropsDemoEditor';
import { Provider as DataContextProvider } from 'Controls-DataEnv/context';
import { RecordSet } from 'Types/collection';
import { actions } from './actions';

function ButtonPropsEditor() {
    const contextConfig = {
        Actions: {
            dataFactoryName: 'Controls/dataFactory:List',
            dataFactoryArguments: {
                keyProperty: 'id',
                displayProperty: 'commandName',
                groupProperty: 'category',
            },
        },
    };

    const contextData = {
        Actions: {
            items: new RecordSet({
                keyProperty: 'id',
                rawData: actions,
            }),
        },
    };
    return (
        <DataContextProvider configs={contextConfig} loadResults={contextData}>
            <PropsDemoEditor metaType={ButtonType} control={Button} />;
        </DataContextProvider>
    );
}

export default ButtonPropsEditor;
