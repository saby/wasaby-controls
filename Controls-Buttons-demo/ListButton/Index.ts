import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import controlTemplate = require('wml!Controls-Buttons-demo/ListButton/Template');
import { RecordSet } from 'Types/collection';
import 'wml!Controls-Buttons-demo/ListButton/resources/template';

export default class ListButton extends Control<IControlOptions> {
    protected _template: TemplateFunction = controlTemplate;
    protected _items: RecordSet = null;
    protected _menuOptions: Object;

    protected _beforeMount(): void {
        this._items = new RecordSet({
            keyProperty: 'id',
            rawData: [
                {
                    id: 0,
                    title: 'Калорийность',
                    additionalText: 'Белки, жиры, углеводы',
                    template: 'wml!Controls-Buttons-demo/ListButton/resources/template',
                },
                {
                    id: 1,
                    title: 'Размеры',
                    additionalText: 'Длинна, ширина, высота',
                    template: 'wml!Controls-Buttons-demo/ListButton/resources/template',
                },
                {
                    id: 2,
                    title: 'Производитель',
                    additionalText: 'Страна, адрес, реквизиты',
                    template: 'wml!Controls-Buttons-demo/ListButton/resources/template',
                },
            ],
        });

        this._menuOptions = {
            items: this._items,
            itemTemplateProperty: 'template',
            keyProperty: 'id',
        };
    }
}
