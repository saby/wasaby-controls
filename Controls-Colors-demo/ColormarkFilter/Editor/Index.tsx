import * as React from 'react';
import { Memory } from 'Types/source';
import { IDataConfig, IListDataFactoryArguments } from 'Controls-DataEnv/dataFactory';
import { View as FilterView } from 'Controls-ListEnv/filterConnected';

const filterItems = [
    {
        id: '4',
        icon: 'Flag',
        iconStyle: 'primary',
        iconSize: 'm',
        caption: 'Важный',
    },
    {
        id: '5',
        icon: 'Add',
        iconStyle: 'primary',
        iconSize: 'm',
        caption: 'Плюс',
    },
    {
        id: '0',
        type: 'color',
        value: {
            color: '--danger_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Красный',
        removable: true,
        editable: true,
    },
    {
        id: '1',
        type: 'color',
        value: {
            color: '--success_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Зеленый',
    },
    {
        id: '2',
        type: 'color',
        value: {
            color: '--secondary_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Синий',
        removable: true,
        editable: true,
    },
    {
        id: '3',
        type: 'color',
        value: {
            color: '--warning_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Желтый',
        removable: true,
        editable: true,
    },
];

export default class extends React.Component {
    static getLoadConfig(): Record<string, IDataConfig<IListDataFactoryArguments>> {
        return {
            markData: {
                dataFactoryName: 'Controls/dataFactory:List',
                dataFactoryArguments: {
                    source: new Memory({
                        data: [],
                        keyProperty: 'id',
                    }),
                    keyProperty: 'id',
                    displayProperty: 'title',
                    filterDescription: [
                        {
                            name: 'mark',
                            editorTemplateName: 'Controls-Colors/colormarkFilter:Editor',
                            resetValue: { selected: [], excluded: [] },
                            viewMode: 'extended',
                            value: { selected: ['1', '3'], excluded: ['0', '2'] },
                            editorOptions: {
                                items: filterItems,
                                extendedCaption: 'Пометки',
                            },
                        },
                    ],
                },
            },
        };
    }

    render() {
        return (
            <div>
                <FilterView ref={this.props.forwardedRef} storeId="markData" />
            </div>
        );
    }
}
