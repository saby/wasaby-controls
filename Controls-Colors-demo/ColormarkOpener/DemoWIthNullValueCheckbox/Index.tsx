import { forwardRef, LegacyRef, ReactElement, useState, useCallback } from 'react';
import { useTheme } from 'UICore/Contexts';
import { itemsWithPartialSelected, palette } from '../../data';
import { DialogOpener } from 'Controls-Colors/colormarkOpener';
import { ItemsView } from 'Controls/list';
import { list } from '../resources/data';
import { Model } from 'Types/entity';
import ItemTemplate from '../templates/ItemTemplate';
import 'css!Controls-Colors-demo/Style';

const helper = new DialogOpener({});
const itemActions = [
    {
        id: 1,
        tooltip: 'Выбрать пометку',
        icon: 'icon-Colorize',
    },
];

export default forwardRef((_: unknown, ref: LegacyRef<HTMLDivElement>): ReactElement => {
    const theme = useTheme();

    const [listMarkedKey, setListMarkedKey] = useState('1');

    const onMarkedKeyChangeHandler = useCallback((key) => {
        setListMarkedKey(key);
    }, []);

    const actionClickHandler = useCallback(
        (_: Event, item: Model<{ keys: string[] }>) => {
            helper.open({
                templateOptions: {
                    items: itemsWithPartialSelected,
                    palette,
                    selectedKeys: item.get('keys') || [],
                    isAdaptive: true,
                    multiSelect: true,
                    onSelectedKeysChanged(keys) {
                        itemsWithPartialSelected.forEach((item) => {
                            if (!keys.includes(item.id)) {
                                item.partialSelected = undefined;
                            }
                        });
                    },
                },
                width: theme === 'default' ? 252 : 405,
                eventHandlers: {
                    onResult(res: string[]) {
                        item.set('keys', res);
                    },
                },
            });
        },
        [theme]
    );

    return (
        <div className="tw-flex tw-justify-center" ref={ref}>
            <div
                className="tw-flex tw-justify-baseline controls-cursor_pointer Controls-Colors-demo_border
                                Controls-Colors-demo_list"
            >
                <ItemsView
                    items={list}
                    className="tw-w-full"
                    itemActions={itemActions}
                    markedKey={listMarkedKey}
                    onActionClick={actionClickHandler}
                    onMarkedKeyChanged={onMarkedKeyChangeHandler}
                    itemTemplate={ItemTemplate}
                    itemTemplateOptions={{ items: itemsWithPartialSelected }}
                />
            </div>
        </div>
    );
});
