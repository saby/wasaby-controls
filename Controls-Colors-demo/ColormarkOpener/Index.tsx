import { forwardRef, LegacyRef, ReactElement, useState, useCallback } from 'react';
import { useTheme } from 'UICore/Contexts';
import { itemsWithCustom, palette } from '../data';
import { DialogOpener } from 'Controls-Colors/colormarkOpener';
import { ItemsView } from 'Controls/list';
import { list } from './resources/data';
import { Model } from 'Types/entity';
import ItemTemplate from './templates/ItemTemplate';
import 'css!Controls-Colors-demo/Style';

const helper = new DialogOpener({});
const itemActions = [
    {
        id: 1,
        tooltip: 'Выбрать пометку',
        icon: 'icon-Colorize',
    },
];

export default forwardRef((_: unknown, ref: LegacyRef<HTMLDivElement>): ReactElement => {
    const theme = useTheme();

    const [listMarkedKey, setListMarkedKey] = useState('1');

    const onMarkedKeyChangeHandler = useCallback((key) => {
        setListMarkedKey(key);
    }, []);

    const actionClickHandler = useCallback(
        (_: Event, item: Model<{ keys: string[] }>) => {
            helper.open({
                templateOptions: {
                    items: itemsWithCustom,
                    palette,
                    selectedKeys: item.get('keys') || [],
                    isAdaptive: true,
                    multiSelect: true,
                },
                width: theme === 'default' ? 252 : 405,
                eventHandlers: {
                    onResult(res: string[]) {
                        item.set('keys', res);
                    },
                },
            });
        },
        [theme]
    );

    return (
        <div className="tw-flex tw-justify-center" ref={ref}>
            <div
                className="tw-flex tw-justify-baseline controls-cursor_pointer Controls-Colors-demo_border
                                Controls-Colors-demo_list"
            >
                <ItemsView
                    items={list}
                    className="tw-w-full"
                    itemActions={itemActions}
                    markedKey={listMarkedKey}
                    onActionClick={actionClickHandler}
                    onMarkedKeyChanged={onMarkedKeyChangeHandler}
                    itemTemplate={ItemTemplate}
                    itemTemplateOptions={{ items: itemsWithCustom }}
                />
            </div>
        </div>
    );
});
