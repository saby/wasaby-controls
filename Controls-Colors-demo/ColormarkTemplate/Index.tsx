import * as React from 'react';
import {Template} from 'Controls-Colors/colormarkTemplate';
import {itemsWithCustom, palette} from '../data';
import 'css!Controls-Colors-demo/Style';

export default React.forwardRef((_: {}, ref: React.LegacyRef<HTMLDivElement>): React.ReactElement => {
        const [selectedKeys, setSelectedKeys] = React.useState(['1']);

        const onSelectedKeysChanged = React.useCallback((keys) => {
            setSelectedKeys(() => keys);
        }, []);

        return (
            <div className="tw-flex tw-justify-center controls-padding_top-m controls-padding_bottom-m"
                 ref={ref}
            >
                <div className="Controls-Colors-demo_widthPanel">
                    <Template items={itemsWithCustom}
                              palette={palette}
                              selectedKeys={selectedKeys}
                              onSelectedKeysChanged={onSelectedKeysChanged}
                    />
                </div>
            </div>
        );
    });
