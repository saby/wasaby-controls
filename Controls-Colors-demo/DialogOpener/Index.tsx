import * as React from 'react';
import {itemsWithCustom, palette} from '../data';
import {DialogOpener} from 'Controls-Colors/dialogOpener';
import 'css!Controls-Colors-demo/Style';

const helper = new DialogOpener({});

export default React.forwardRef(
    (props, ref: React.LegacyRef<HTMLDivElement>): React.ReactElement => {
        const [selectedKeys, setSelectedKeys] = React.useState(['3']);
        const itemsContainer: React.MutableRefObject<HTMLDivElement> = React.useRef();
        const getColor = React.useCallback((item) => {
            const value = item.value.color;
            return value.indexOf('--') !== -1 ? 'var(' + value + ')' : value;
        }, []);
        const itemsClickHandler = React.useCallback(() => {
            helper.open({
                opener: itemsContainer.current,
                templateOptions: {
                    items: itemsWithCustom,
                    palette,
                    selectedKeys
                },
                width: props.theme === 'default' ? 252 : 405,
                eventHandlers: {
                    onResult(res: string[]) {
                        setSelectedKeys(res);
                    }
                }
            });
        },[selectedKeys]);

        React.useEffect(() => {
            return () => helper.destroy();
        }, []);

        return (
            <div className="tw-flex tw-justify-center"
                 ref={ref}
            >
                <div className="tw-flex tw-justify-baseline controls-cursor_pointer"
                     ref={itemsContainer}
                     onClick={itemsClickHandler}
                >
                    {
                        selectedKeys.length > 0 ? itemsWithCustom.map((item) => {
                            if (selectedKeys.includes(item.id)) {
                                return (
                                    <div className="controls-margin_right-s tw-self-center"
                                         key={item.id}
                                    >
                                        {
                                            item.type === 'color' ? (
                                                    <div className="Controls-Colors-demo_roundItem"
                                                         style={{backgroundColor: getColor(item)}}
                                                    >
                                                    </div>
                                                ) :
                                                (
                                                    item.icon ? (
                                                        <div className={'Controls-Colors-demo_iconItem icon-' + item.icon +
                                                            ' controls-icon_style-' + item.iconStyle + ' controls-icon_size-' +
                                                            item.iconSize
                                                        }>
                                                        </div>
                                                    ) : (
                                                        <div style={{
                                                            color: getColor(item),
                                                            fontWeight: item.value.style.b ? 'bold' : 'normal',
                                                            fontStyle: item.value.style.i ? 'italic' : 'normal',
                                                            textDecoration: item.value.style.s ? 'line-through' : 'none',
                                                            borderBottom: item.value.style.u ? ('1px solid ' + getColor(item)) : ''
                                                        }}>
                                                            {item.caption}
                                                        </div>
                                                    )
                                                )
                                        }
                                    </div>
                                );
                            }
                        }) :
                        <div className='controls-text-label'>Выбрать пометку</div>
                    }
                </div>
            </div>
        );
    });
