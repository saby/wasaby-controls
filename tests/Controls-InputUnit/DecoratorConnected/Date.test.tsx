/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';
import { getContent } from '../resources/utils';
import { default as Index } from 'Controls-Input-demo/DecoratorConnected/Date/Index';
import { default as Mask } from 'Controls-Input-demo/DecoratorConnected/Date/Mask';

describe(
    'Controls-Input/decoratorConnected:Date',
    () => {
        let container = null;

        beforeEach(() => {
            jest.useFakeTimers('modern');
            jest.setSystemTime(new Date(1996, 5, 21));
            container = document.createElement('div');
            WasabyEvents.initInstance(container);
            document.body.appendChild(container);
        });
        afterEach(() => {
            unmountComponentAtNode(container);
            WasabyEvents.destroyInstance(container);
            jest.useRealTimers();
            container.remove();
            container = null;
        });

        it('Index', async () => {
            render(getContent(Index), {container});
            expect(container).toMatchSnapshot();
        });
        it('Mask', async () => {
            render(getContent(Mask), {container});
            expect(container).toMatchSnapshot();
        });
    }
);
