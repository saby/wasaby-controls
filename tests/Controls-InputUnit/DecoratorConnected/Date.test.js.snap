// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/decoratorConnected:Date Index 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <span
      class="controls-DecoratorDate controls-fontsize-undefined
                         controls-text-undefined
                         controls-fontweight-undefined"
      title="02.02.22"
    >
      02.02.22
    </span>
  </div>
</div>
`;

exports[`Controls-Input/decoratorConnected:Date Mask 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="controls-margin_bottom-m"
    >
      <span
        class="controls-DecoratorDate controls-fontsize-undefined
                         controls-text-undefined
                         controls-fontweight-undefined"
        title="02.02.2022"
      >
        02.02.2022
      </span>
    </div>
    <div
      class="controls-margin_bottom-m"
    >
      <span
        class="controls-DecoratorDate controls-fontsize-undefined
                         controls-text-undefined
                         controls-fontweight-undefined"
        title="02.02.22"
      >
        02.02.22
      </span>
    </div>
    <div
      class="controls-margin_bottom-m"
    >
      <span
        class="controls-DecoratorDate controls-fontsize-undefined
                         controls-text-undefined
                         controls-fontweight-undefined"
        title="02.2022"
      >
        02.2022
      </span>
    </div>
    <div
      class="controls-margin_bottom-m"
    >
      <span
        class="controls-DecoratorDate controls-fontsize-undefined
                         controls-text-undefined
                         controls-fontweight-undefined"
        title="2022"
      >
        2022
      </span>
    </div>
  </div>
</div>
`;
